### PR TITLE
PLTFRS-11334 RIP node-tech-time

### PR DESCRIPTION
**Pull request by jlescot**
Fix https://issues.sierrawireless.com/browse/PLTFRS-11334

#### Changes
- Remove also `node-tech-ruuid` dependency
- Replace `var` to `const`/`let`
- Use arrow functions
- Use some shorthand notations
- Add a `yarn.lock` file

:warning: The `on: emitter.on.bind(emitter)` has been removed... we need to make sure that it is no more called by the webapps relying on this module.

#### Smoke Tests
Pre-requisites: Launch `av-accounts` locally, targeting this branch and using a local MySQL DB

- [ ] **DB migrations**
  - `yarn migrate status`
  - `yarn migrate down 1`
  - `yarn migrate up 1`
- [ ] **APIs**
  - Navigate into different pages: List subaccounts, update a subaccount, create a token operation, edit a token operation, remove a token operation...

### CODE REVIEW

File: lib/wrapSql.js [hunk 1]
```diff
@@ -1,90 +1,25 @@
...
-
-      return wrapSqlFn(
-        ruuid,
-        function() {
-          return originalFn.apply(model, originalFnArgs);
-        },
-        category
-      );
+  const wrapByName = fnName => {
+    const wrappedFunction = (ruuid, ...originalFnArgs) => {
```
Comment (adriffaud) [inline lib/wrapSql.js:15]: :heart_eyes:

**Review by adriffaud — APPROVED** [review]
LGTM :+1:

File: lib/lockService.js [hunk 2]
```diff
@@ -8,148 +8,126 @@ var BPromise = require('bluebird');
...
+   *
+   * @param  {String}   name              The name of the lock to acquire
+   * @param  {Object}   options           See below for details description
+   * @param  {Integer}  options.attempt   The number of attempt to acquire a lock. Default value: 1.
+   * @param  {Integer}  options.delay     The delay between 2 attempts. Default value: 500ms.
+   * @param  {Integer}  options.timeout   The number of seconds where the lock is considered as expired. Default value: 60s.
+   *
+   * @return Promise
+   */
+  const _lock = (name, options) => {
```
Comment (sgaestel) [inline lib/lockService.js]: Possible optimization:
```
const _lock = (name, {maxAttempt=1, delay=500, timeout=60}) => {
  return ...
}
```

File: lib/migration.js [hunk 3]
```diff
@@ -17,18 +17,12 @@ var schemaVersion = require('./schemaVersion');
...
-module.exports = function(options) {
-  var sequelize = options.sequelize;
-  var Sequelize = options.Sequelize;
-  var lockService = options.lockService;
-
-  var migrationCfg = options.config || {};
-  var umzug = new Umzug({
+module.exports = options => {
+  const { sequelize, Sequelize, lockService } = options;
+  const migrationCfg = options.config || {};
```
Comment (sgaestel) [inline lib/migration.js]: Possible optimization:
```
module.exports = ({sequelize, Sequelize, lockService, config={}}) => {
```

**Review by sgaestel — APPROVED** [review]
:+1: Code LGTM
Only 2 potential optimizations, but really nice to have.

File: lib/lockService.js (hunk 2 above)
Comment (jlescot) [reply lib/lockService.js]: > I will then keep the option named `attempt` instead of `maxAttempt` 😉

**Pull request merged**